mgmt2/rpc/handlers/common/RecordsUtils.cc

```diff
@@ -0,0 +1,313 @@
+{
+  Context ctx;
+
+  ctx.checkCb = [](RecT rec_type, std::error_code &ec) -> bool {
+    if (!REC_TYPE_IS_CONFIG(rec_type)) {
+      // we have an issue
+      ec = rpc::handlers::errors::RecordError::GENERAL_ERROR;
+      return false;
+    }
+    return true;
+  };
+
+  // send this back.
+  get_record_impl(name, ctx);
+
+  return {ctx.yaml, ctx.ec};
+}
+
+void static get_record_regex_impl(std::string_view name, unsigned recType, Context &ctx)
+{
+  auto yamlConverter = [](const RecRecord *record, void *data) {
+    auto &ctx = *static_cast<Context *>(data);
+
+    if (!record) {
+      return;
+    }
+
+    YAML::Node recordYaml;
+    // convert it.
+    try {
+      recordYaml = *record;
+    } catch (std::exception const &ex) {
+      ctx.ec = rpc::handlers::errors::RecordError::GENERAL_ERROR;
+      return;
+    }
+
+    // we have to append the records to the context one.
+    ctx.yaml.push_back(recordYaml);
+  };
+
+  const auto ret = RecLookupMatchingRecords(recType, name.data(), yamlConverter, &ctx);
+  if (ctx.ec) {
+    return;
+  }
+
+  if (ret != REC_ERR_OKAY) {
+    // append
+    Error("We couldn't get the records that match '%s'", name.data());
+    ctx.ec = rpc::handlers::errors::RecordError::GENERAL_ERROR;
+    return;
+  }
+}
+
+std::tuple<YAML::Node, std::error_code>
+get_yaml_record_regex(std::string_view name, unsigned recType)
+{
+  Context ctx;
+
+  get_record_regex_impl(name, recType, ctx);
+
+  return {ctx.yaml, ctx.ec};
+}
+
+// Basic functions to helps setting a record value properly. All this functions were originally in the WebMgmtUtils.
+namespace
+{ // anonymous namespace
+  bool
+  recordRegexCheck(std::string_view pattern, std::string_view value)
+  {
+    pcre *regex;
+    const char *error;
+    int erroffset;
+
+    regex = pcre_compile(pattern.data(), 0, &error, &erroffset, nullptr);
+    if (!regex) {
+      return false;
+    } else {
+      int r = pcre_exec(regex, nullptr, value.data(), value.size(), 0, 0, nullptr, 0);
+
+      pcre_free(regex);
+      return (r != -1) ? true : false;
+    }
+
+    return false; // no-op
+  }
+
+  bool
+  recordRangeCheck(std::string_view pattern, std::string_view value)
+  {
+    char *p = const_cast<char *>(pattern.data());
+    Tokenizer dashTok("-");
+
+    if (recordRegexCheck("^[0-9]+$", value)) {
+      while (*p != '[') {
+        p++;
+      } // skip to '['
+      if (dashTok.Initialize(++p, COPY_TOKS) == 2) {
+        int l_limit = atoi(dashTok[0]);
+        int u_limit = atoi(dashTok[1]);
+        int val     = atoi(value.data());
+        if (val >= l_limit && val <= u_limit) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  bool
+  recordIPCheck(std::string_view pattern, std::string_view value)
+  {
+    //  regex_t regex;
+    //  int result;
+    bool check;
+    std::string_view range_pattern = R"(\[[0-9]+\-[0-9]+\]\\\.\[[0-9]+\-[0-9]+\]\\\.\[[0-9]+\-[0-9]+\]\\\.\[[0-9]+\-[0-9]+\])";
```

Comment:
You should either use existing support in "ink_inet.h" or add this if it's not there. One change that could be done is to enable the address parsing to take a `nullptr` for `sockaddr*` to indicate only parsing is to be done, not storing, for exactly this kind of validity check.

Reply:
This is code from TM -> https://github.com/brbzull0/trafficserver/blob/25d13c1b33117d1969971fa4c10b55077438d880/mgmt/WebMgmtUtils.cc#L1063-L1071

I may review this and see if we can use a different(or existing) approach.